### ISSUE TITLE
[BUG] It seems that the model parameters are not updated.

### ISSUE THREAD

**Issue by sooftware** [microsoft/DeepSpeed#1483]
**Describe the bug**
I use DeepSpeed with GPT-Neo 2.7B model. I success started training, but it seems that the model parameters are not updated.  
  
<img src="https://user-images.githubusercontent.com/42150335/138831336-d98e7ad1-7394-4fa5-bd62-bebd0050475c.png" width=400>
  
Below is my code and deepspeed config. If there is anything strange or suspicious, please tell me.
  
```python
...
...

torch.distributed.init_process_group(backend="nccl")
deepspeed.init_distributed("nccl")

model = GPTNeoForCausalLM.from_pretrained(args.pretrain_model_name_or_path).to(device)
model.resize_token_embeddings(len(tokenizer))
wandb.watch(model)

config = {
    "train_batch_size": 1,
    "fp16": {
        "enabled": True,
    },
    "zero_optimization": {
        "stage": 2,
        "contiguous_gradients": True,
        "overlap_comm": True,
        "offload_optimizer": {
            "device": "cpu",
            "pin_memory": True,
            "fast_init": True
        },
    },
    "optimizer": {
        "type": "Adam",
        "params": {
            "lr": 3e-05,
            "betas": [
                0.9,
                0.999
            ],
            "eps": 1e-8
        }
    },
}

model, optimizer, _, _ = deepspeed.initialize(model=model,
                                              config_params=config,
                                              model_parameters=model.parameters())

for step, (inputs, attention_masks, labels, answers) in enumerate(train_loader):
    inputs = inputs.to(self.device)
    attention_masks = attention_masks.to(self.device)
    labels = labels.to(self.device)

    outputs = self.model(input_ids=inputs, attention_mask=attention_masks, labels=labels)
    loss = outputs.loss

    model.backward(loss)
    model.step()

...
...
```

Training proceeds. The problem is that the loss does not decrease at all, so it seems that the parameter update is not possible. If I train with a small model without applying deep speed, the loss falls well.

**System info (please complete the following information):**
 - OS: Ubuntu 20.04
 - GPU count and types [e.g. two machines with x8 A100s each]: A100 x 1
 - Python version: 3.8

**Comment by sooftware** [comment]
Never mind. It seems to be a problem with my equipment.

**Comment by gouqi666** [comment]
hello, I met this problem too, could u share your solution? @sooftware

**Comment by sooftware** [comment]
Sorry. I don't remember about this issue.
Maybe environmental problem I guess.
